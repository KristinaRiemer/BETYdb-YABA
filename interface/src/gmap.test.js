import React from "react"
import { shallow } from 'enzyme'
import GMap from './components/gmap'

describe('Test the GMap Component',()=>{
    it('it should render correctly',()=>{
        const gmap = shallow(<GMap />, {
            disableLifecycleMethods: true
        });
        const geoJSON= {
            "type": "FeatureCollection",
             "features": [
               { "type": "Feature",
                 "geometry": {"type": "Point", "coordinates": [102.0, 0.5]},
                 "properties": {"prop0": "value0"}
                 },
               { "type": "Feature",
                 "geometry": {
                   "type": "LineString",
                   "coordinates": [
                     [102.0, 0.0], [103.0, 1.0], [104.0, 0.0], [105.0, 1.0]
                     ]
                   },
                 "properties": {
                   "prop0": "value0",
                   "prop1": 0.0
                   }
                 },
               { "type": "Feature",
                 "geometry": {
                   "type": "Polygon",
                   "coordinates": [
                     [ [100.0, 0.0], [101.0, 0.0], [101.0, 1.0],
                       [100.0, 1.0], [100.0, 0.0] ]
                     ]
           
                 },
                 "properties": {
                   "prop0": "value0",
                   "prop1": {"this": "that"}
                   }
                 }
               ]
             };
        const bounds=[[0, 100],[1, 105]];
        gmap.setState({
            geoJSON,
            bounds
        })

        expect(gmap).toMatchSnapshot();
    })
})