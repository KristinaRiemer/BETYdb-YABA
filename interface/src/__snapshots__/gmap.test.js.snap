// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the GMap Component it should render correctly 1`] = `
<Map
  animate={true}
  attributionControl={true}
  bounds={
    Array [
      Array [
        0,
        100,
      ],
      Array [
        1,
        105,
      ],
    ]
  }
  center={
    Array [
      51.505,
      -0.09,
    ]
  }
  doubleClickZoom={true}
  dragging={true}
  easeLinearity={0.35}
  scrollWheelZoom={true}
  touchZoom={true}
  zoomControl={true}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="&copy <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.osm.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(GeoJSON))
    data={
      Object {
        "features": Array [
          Object {
            "geometry": Object {
              "coordinates": Array [
                102,
                0.5,
              ],
              "type": "Point",
            },
            "properties": Object {
              "prop0": "value0",
            },
            "type": "Feature",
          },
          Object {
            "geometry": Object {
              "coordinates": Array [
                Array [
                  102,
                  0,
                ],
                Array [
                  103,
                  1,
                ],
                Array [
                  104,
                  0,
                ],
                Array [
                  105,
                  1,
                ],
              ],
              "type": "LineString",
            },
            "properties": Object {
              "prop0": "value0",
              "prop1": 0,
            },
            "type": "Feature",
          },
          Object {
            "geometry": Object {
              "coordinates": Array [
                Array [
                  Array [
                    100,
                    0,
                  ],
                  Array [
                    101,
                    0,
                  ],
                  Array [
                    101,
                    1,
                  ],
                  Array [
                    100,
                    1,
                  ],
                  Array [
                    100,
                    0,
                  ],
                ],
              ],
              "type": "Polygon",
            },
            "properties": Object {
              "prop0": "value0",
              "prop1": Object {
                "this": "that",
              },
            },
            "type": "Feature",
          },
        ],
        "type": "FeatureCollection",
      }
    }
  />
</Map>
`;
